### ISSUE TITLE
Support builds for esp8266

### ISSUE THREAD

**Issue by hmaerki** [mattytrentini/mpbuild#53]
Damien would like to test esp8266 on octoprobe.

So I will have a quick look if I can create a container.
I try to create a container based on https://github.com/micropython/micropython/tree/master/ports/esp8266#readme which refers to this container: https://github.com/larsks/docker-image-esp-open-sdk.

If there is another reason why this build is not supported by mpbuild - please let me know.

**Comment by mattytrentini** [comment]
No, the only reason it's not supported is because I haven't created a handy container. However, I believe @andrewleech has used one in the past, not sure if it's up-to-date. I've asked Andrew to chime in...

**Comment by andrewleech** [comment]
Ah yes, this is my old container: 
registry.gitlab.com/alelec/docker-esp-open-sdk

The project to build it is: https://gitlab.com/alelec/docker-esp-open-sdk

I'm not sure if it needs updating, has the SDK for that chip been updated in recent years?

**Comment by Josverl** [comment]
im a user of containers , rather than a builder ,

but it seems to me that it may be good to (re-)use the maintenance effort that is put into https://github.com/Josverl/micropython/blob/master/tools/ci.sh to build a container.

I use that myself to build (pne-off) containers for codespaces , mostly so I do not need to spend too much time backtracking the differences between documentation and reality, and that seems to work well.
one downside is that - unless I use prebuilt containers - I need to drink some coffee - still way less c☕☕than the manual approach.

Woud that also work to create containers for the different ports ?

**Comment by mattytrentini** [comment]
ESP8266 builds are now supported (thanks to the good work of @hmaerki!) in v0.6 of mpbuild.
